packages/teleport/src/Player/DesktopPlayer.tsx

```diff
@@ -65,7 +77,39 @@ export const ProgressBarDesktop = (props: { playerClient: PlayerClient }) => {
       });
     });
 
+    const throttledUpdateCurrentTime = throttle(
+      currentTimeMs => {
+        setState(prevState => {
+          return {
+            ...prevState,
+            current: currentTimeMs,
+            time: toHuman(currentTimeMs),
+          };
+        });
+      },
+      // Magic number to throttle progress bar updates so that the playback is smoother.
+      50
+    );
+
+    playerClient.addListener(
+      PlayerClientEvent.UPDATE_CURRENT_TIME,
+      currentTimeMs => throttledUpdateCurrentTime(currentTimeMs)
+    );
+
+    playerClient.addListener(PlayerClientEvent.SESSION_END, () => {
+      throttledUpdateCurrentTime.cancel();
+      // TODO: This logic can cause the progress bar to jump abruptly to the end, if the session was sitting without
```

Comment:
Usually I create issues and then link them in the code along side TODO's so that they don't get lost.